### PR TITLE
Alvarocabanas/static analysis

### PR DESCRIPTION
**Pull request by alvarocabanas**

### CODE REVIEW

File: .github/workflows/prerelease.yml [hunk 1]
```diff
@@ -16,19 +16,18 @@ env:
   TAG: ${{ github.event.release.tag_name }}
 
 jobs:
-
-  validate:
-    name: Validate code via linters
+  static-analysis:
```
Comment (ivancorrales) [inline .github/workflows/prerelease.yml:19]: I think we should change the reference to `validate`  in  the below block `needs`

`validate` should be replaced by `static-control`

File: Makefile [hunk 2]
```diff
@@ -18,37 +19,26 @@ clean:
...
-
-compile: deps
-	@echo "=== $(INTEGRATION) === [ compile ]: Building $(BINARY_NAME)..."
-	@go build -o bin/$(BINARY_NAME) ./src
+validate:
+ifeq ($(strip $(GO_FILES)),)
+	@echo "=== $(INTEGRATION) === [ validate ]: no Go files found. Skipping validation."
+else
+	@printf "=== $(INTEGRATION) === [ validate ]: running golangci-lint & semgrep... "
+	@go run  $(GOFLAGS) github.com/golangci/golangci-lint/cmd/golangci-lint run --verbose
```
Comment (ivancorrales) [inline Makefile:27]: I think we could use variable $(GOLANGCI_LINT)

```bash
@go run  $(GOFLAGS) $(GOLANGCI_LINT) run --verbose
```

**Review by ivancorrales — CHANGES_REQUESTED** [review]

File: Makefile [hunk 3]
```diff
@@ -18,37 +19,26 @@ clean:
...
-	@echo "=== $(INTEGRATION) === [ compile ]: Building $(BINARY_NAME)..."
-	@go build -o bin/$(BINARY_NAME) ./src
+validate:
+ifeq ($(strip $(GO_FILES)),)
+	@echo "=== $(INTEGRATION) === [ validate ]: no Go files found. Skipping validation."
+else
+	@printf "=== $(INTEGRATION) === [ validate ]: running golangci-lint & semgrep... "
+	@go run  $(GOFLAGS) $(GOLANGCI_LINT) run --verbose
+	@if [ -f .semgrep.yml ]; then \
+        docker run --rm -v "${PWD}:/src:ro" --workdir /src returntocorp/semgrep -c .semgrep.yml ; \
```
Comment (gsanchezgavier) [inline Makefile:29]: it would have been nice that semgrep was done in Go so we add it to the tools.go  😞

**Review by gsanchezgavier — APPROVED** [review]
lgtm just a small comment on the makefile, thanks!

**Pull request merged**